### ISSUE TITLE
Make it extra-clear to users that data is never uploaded

### ISSUE THREAD

**Issue by JulienVig** [epfml/disco#1137]
Users are still getting confused about whether their confidential data is being uploaded online. We should make it unambiguous and intuitive that data always stay local and is only read by the browser when training data locally. Some ideas:
- Instant feedback: The moment files are selected, show parsed metadata immediately: row counts, column names, file sizes, a preview. Anything that appears faster than an upload could possibly complete: "Read 48,210 rows locally in 0.3s."
- Live data transfer counter: "Data uploaded: 0 bytes." Then, when training rounds send gradients, the counter shows exactly what left: "Model updates sent: 2.1 MB (weights only — no raw data)."
- data stays local icon with secure icon
- suggest to users to test it by turning off their wifi when connecting the data (though it doesn't prove that data is not uploaded later during training) or checking the devtools network tab

Mock up:

<img width="601" height="489" alt="Image" src="https://github.com/user-attachments/assets/cdd4677d-3f0d-4cd5-9c89-8f794aa45bcd" />

**Comment by hleepfl** [comment]
I like the instant feedback and lock icon ideas !
The data transfer counter is a dangerous one I think, I can see questions of "why are you mentionning uploaded data if there isn't any ?"

2 more possible ideas
- Keep the interface as minimal as possible (in this regard I like the current interface, but even the Ctrl + A tip could be removed) so that the user can only focus on the most important thing (nothing is uploaded). I do like the suggestion to turn off wifi, but it could be moved in a tooltip to keep visual elements to the bare minimum.
- Once the files are selected, display them as file system paths to reinforce the notion of files being local

Edit: Other mock up
<img width="428" height="305" alt="Image" src="https://github.com/user-attachments/assets/93f5c7ca-5642-4cea-ad05-5550a2222cc2" />

**Comment by JulienVig** [comment]
Completely agree!

**Comment by hleepfl** [comment]
Okay I just learned that paths are not accessible in the browser
What do we think of showing a fake path to make it visually look like a local filesystem ? I don't know if that would just add more confusion

**Comment by JulienVig** [comment]
Sounds too confusing I think just the filenames will be enough
